src/AbstractEventManager.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace Dhii\WpEvents;
+
+use Psr\EventManager\EventInterface;
+use Dhii\WpEvents\Exception\ExceptionInterface;
+
+/**
+ * Common functionality for event managers.
+ *
+ * @since [*next-version*]
+ */
+abstract class AbstractEventManager
+{
+    const DEFAULT_PRIORITY = 10;
+
+    /**
+     * Parameter-less constructor.
+     * 
+     * @since [*next-version*]
+     */
+    protected function _construct()
+    {
+    }
+
+    /**
+     * Normalizes the given event into an Event instance.
+     *
+     * @since [*next-version*]
+     *
+     * @param EventInterface|string $event  Event instance or an event name string.
+     * @param object                $target The target of the event.
+     * @param array                 $argv   Arguments for the event.
+     *
+     * @return EventInterface The event instance.
+     */
+    protected function _normalizeEvent($event, $target = null, $argv = array())
+    {
+        $normalizedEvent = $event instanceof EventInterface
+                ? $event
+                : $this->_createEvent($event, $argv, $target);
+        /* @var $normalizedEvent \Psr\EventManager\EventInterface */
+        $normalizedEvent->setParams($this->_mergeArgs($normalizedEvent->getParams(), $argv));
+
+        return $normalizedEvent;
+    }
+
+    /**
+     * Merges values from an array into a base array, overwriting values irrespective of type.
+     *
+     * @since [*next-version*]
+     *
+     * @param array $base  The array to merge into.
+     * @param array $other The array to merge from.
+     *
+     * @return array The result of merging the arrays.
+     */
+    protected function _mergeArgs($base, $other)
```

Comment:
Forgive me if this is a stupid question, but is there any particular reason for not using `array_merge()`?